CHANGELOG.md

```diff
@@ -9,6 +9,29 @@ stages](https://github.com/kubernetes/community/blob/master/contributors/devel/s
 
 ## Unreleased
 
+## 0.4.0 - 2019-02-20
+
+### Added
+
+- Add support for new encapsulation modes: noEncap (inter-Node Pod traffic is never encapsulated) and hybrid (inter-Node Pod traffic is only encapsulated if Nodes are not on the same subnet).
+- Add support for "named ports" to Network Policy implementation.
+- Add user documentation for IPsec support.
+- Add support for new "table" output mode to antctl.
+
+### Changed
+
+- Changes in OpenFlow client:
+   * use OpenFlow "bundle" to install related Network Policy flows as part of the same transaction
+   * use flow-based tunelling even when IPsec encryption is enabled
+- Use patched OVS version in Antrea Docker image to avoid cleaning-up datapath flows on graceful antrea-ovs container exit.
+- Reduce amount of Antrea Controller logs when computing Network Policies.
+
+### Fixed
+
+- Fix bug in the Agent that caused some Network Policies not to be enforced properly: for some flows the agent would overwrite existing conjunctive actions instead of adding new actions to the existing ones. This can notably happen when using a /32 ipBlock CIDR to select sources / destinations.
```

Comment:
would like to confirm with @wenyingd @tnqn that this was not introduced within the  0.4.0 cycle as a by product of named port support?

Reply:
This issue exists because there is a fault in the design on Antrea Openflow conjunctive match flow context.

It is about the difference between Antrea and OVS.  Antrea agent has maintained a cache for OpenFlow conjunctive matching entries. Originally, agent thought IP and IP block as two different objects, so it maintains two records in the cache. But OVS always looks IP/32 and IP as the same object, so there should be only one flow entry on OVS. But since Antrea agent has two contexts, and it doesn't merge them, the result is the latter flow would overwrite the previous one.

Reply:
Thanks @wenyingd. Leaving it in release notes since 0.3.0 was affected.